collection-views.js

```diff
@@ -90,13 +117,56 @@ CollectionView.prototype._mutateSelector = function (selector, query) {
       // If narrowingQuery is a function we should pass the result of calling that function
       if (_.isFunction(narrowingQuery))
         narrowingQuery = narrowingQuery();
-      _.extend(query, narrowingQuery)
+      _.extend(query, narrowingQuery);
     }
-    collection = collection._parentCollection
+    collection = collection._parentCollection;
   }
 
   if (LocalCollection._selectorIsId(selector))
     selector = {_id: selector};
 
   return _.extend({}, selector, query);
+};
+
+/**
+ * @summary This function builds the options object by extending the parent's options object
+ * @locus Anywhere
+ * @protected
+ * @method _mutateOptions
+ * @memberOf CollectionView
+ * @param {Object | String} The input options
+ * @return {Object} The chained options object
+ */
+CollectionView.prototype._mutateOptions = function (options) {
+  var collection = this
+    , options = options || {};
+
+  while (collection) {
+    if (! _.isUndefined(collection._narrowingOptions)) {
+      _.each(collection._narrowingOptions, function (val, key) {
+        if (! _.isUndefined(options[key]))
+          _.extend(options[key], val);
+        else
+          options[key] = val;
+      });
+    }
+    
+    collection = collection._parentCollection;
+  }
+
+  // mix including and excluding fields
+  var removeExcludingFields = !! _.find(options.fields, function(val, key) {
+    if (val === 1) {
+      return true;
+    }
+  });
+
+  if (removeExcludingFields) {
+    _.each(options.fields, function (val, key) {
+      if (val === 0)
+        delete options.fields[key];
+    });
+  }
```

Comment:
This code is good, excluding fields is meaningless if we have a field include, because a field include implicitly excludes all non-included fields. However we need to be doing this inside of the loop above, and with an additional check:

```
// this code runs inside of the while loop which continuously narrows the collection
// currently the loop runs in reverse order, which should work if you set your variables correctly
var outerFields, innerFields; // set these variables appropriately
var remainingFields = {};
_.each(innerFields, function (a, key) {
  if (a == 1 && (!_.any(outerFields, isOne) || outerFields[key] == 1)) {
    remainingFields[key] = a;
  } else if (a == 0 && !_.any(innerFields, isZero) && !_.any(outerFields, isZero)) {
    remainingFields[key] = a;
  }
});
_.each(outerFields, function (a, key) {
  if (a == 0) {
    remainingFields[key] = 0;
  }
});
// at this point remove excluding fields if there are including fields
```